### ISSUE TITLE
Children of sealed class are not resolved properly in version 3.1.0

### ISSUE THREAD

**Issue by pallavijoshi44** [InsertKoinIO/koin#1126]
**Describe the bug**

CheckModulesTest fails to resolve the child of a sealed class and throws the following exception: 

```
org.koin.core.error.BrokenDefinitionException: Definition is broken: [Factory:'com.koin.test.SealedClassParameterReceivingClass'] 
- due to error: org.koin.core.error.DefinitionParameterException: No value found for type 'com.koin.test.ParentClass$ChildClass'

```
**To Reproduce**

Inject dependency with parameters like this: 

```
val sealedClassParameterReceivingClass: SealedClassParameterReceivingClass by inject { parametersOf(ParentClass.ChildClass) }
```
Provide dependency like this: 
```
factory { SealedClassParameterReceivingClass(it.get()) }
```
Class declaration : 

```
class SealedClassParameterReceivingClass(private val parentClass: ParentClass.ChildClass) 

sealed class ParentClass {

    object ChildClass : ParentClass() 
}

```
Run CheckModulesTest 

Here is the content of the test: 
```
@Category(CheckModuleTest::class)
class CheckModulesTestWithWorkaround : KoinTest {

    @get:Rule
    val mockProvider = MockProviderRule.create { clazz ->
        mockkClass(clazz)
    }

    @Test
    fun checkAllModules() {
        checkModules(
            allowedMocks = listOf(
                ParentClass::class
            )
        ) {
            androidContext(mockk())
            modules(appModule)
        }
        stopKoin()
    }
}

```
**Expected behavior**
The test should pass.

**Koin project used and used version**
`koin-core version 3.1.0`

**Workaround**
By providing the child class in the allowedMocks, the test passes. 

```
@Category(CheckModuleTest::class)
class CheckModulesTestWithWorkaround : KoinTest {

    @get:Rule
    val mockProvider = MockProviderRule.create { clazz ->
        mockkClass(clazz)
    }

    @Test
    fun checkAllModules() {
        checkModules(
            allowedMocks = listOf(
                ParentClass.ChildClass::class
            )
        ) {
            androidContext(mockk())
            modules(appModule)
        }
        stopKoin()
    }
}
```

Can you please check if this can be fixed as providing the parent class should work for children too.

**Comment by arnaudgiuliani** [comment]
Fixed back in 3.1.1

**Comment by SaeedRe** [comment]
I have the same problem when using Koin-Annotations v1.4.0-RC3 in a KMP project.

```
package com.example.app

sealed class SealedClass {

     @Single(binds = [childSealedClass1::class])
     class childSealedClass1 : SealedClass() {
     } 

     @Single(binds = [childSealedClass2::class])
     class childSealedClass2 : SealedClass() {
     } 

}

```

```
@Module
@ComponentScan("com.example.app")
class MainModule {
}
```

Generated module with KSP:
```
public val com_example_app_MainModule : Module get() = module {
     single()  {com.example.app.childSealedClass1()}bind(com.example.app.SealedClass.childSealedClass1::class)
}

```
Error: Unresolved reference 'childSealedClass1'.
